sound/soc/sof/sof-audio.h

```diff
@@ -30,6 +31,69 @@
 
 #define WIDGET_IS_DAI(id) ((id) == snd_soc_dapm_dai_in || (id) == snd_soc_dapm_dai_out)
 
+struct snd_sof_widget;
+
+/*
+ * Topology Token Parsing.
+ * New tokens should be added to headers and parsing tables below.
+ */
+
+struct sof_topology_token {
+	u32 token;
+	u32 type;
+	u32 (*get_token)(void *elem, void *object, u32 offset, u32 size);
+	u32 offset;
+	u32 size;
+};
+
+struct snd_sof_tuple {
+	u32 token;
+	union {
+		u32 v;
+		const char *s;
+	} value;
+};
+
+extern const struct sof_topology_token pcm_tokens[];
+extern const struct sof_topology_token pipeline_tokens[];
+extern const struct sof_topology_token sched_tokens[];
+extern const struct sof_topology_token asrc_tokens[];
+extern const struct sof_topology_token src_tokens[];
+extern const struct sof_topology_token comp_tokens_new[];
+extern const struct sof_topology_token buffer_tokens[];
+extern const struct sof_topology_token volume_tokens[];
+extern const struct sof_topology_token process_tokens[];
+extern const struct sof_topology_token dai_tokens[];
+extern const struct sof_topology_token dai_link_tokens[];
+extern const struct sof_topology_token hda_tokens[];
+extern const struct sof_topology_token ssp_tokens[];
+extern const struct sof_topology_token alh_tokens[];
+extern const struct sof_topology_token dmic_tokens[];
+extern const struct sof_topology_token dmic_pdm_tokens[];
+extern const struct sof_topology_token esai_tokens[];
+extern const struct sof_topology_token sai_tokens[];
+extern const struct sof_topology_token afe_tokens[];
+extern int pcm_token_size;
+extern int comp_token_size;
+extern int tone_token_size;
+extern int pipeline_token_size;
+extern int sched_token_size;
+extern int buffer_token_size;
+extern int src_token_size;
+extern int asrc_token_size;
+extern int volume_token_size;
+extern int process_token_size;
+extern int dai_token_size;
+extern int dai_link_token_size;
+extern int ssp_token_size;
+extern int alh_token_size;
+extern int hda_token_size;
+extern int dmic_token_size;
+extern int dmic_pdm_token_size;
+extern int esai_token_size;
+extern int sai_token_size;
+extern int afe_token_size;
```

Comment:
Lots of externs...

They present in `topology.c`
These are IPC3 struct descriptions.
They are used directly like exported in `ipc3-topology.c` and via `struct snd_sof_token_desc` in `topology.c`
The `snd_sof_token_desc` is grouping `struct sof_topology_token`s for a given `enum snd_soc_dapm_type` and in `topology.c` we only use the `sof_topology_token.type` so we know what is the expected type the tuple should be as we pass NULL for the object to not store the data away in a struct.

Would it make sense to move these to `ipc3-topology.c` and store them `struct ipc_tplg_ops` (probably rename the struct to ipc_tplg_data?) and in `topology.c` just access it via array to reduce the exports and dependency?